src/components/calculator.js

```diff
@@ -1,71 +1,68 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import calculate from '../logic/calculate';
 import Answer from './Answer';
 import './css/calculator.css';
 
-export default class Calculator extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      total: 0,
-      next: null,
-      operation: null,
-    };
-  }
+const Calculator = () => {
+  const [state, callBack] = useState({
+    total: '0',
+    next: null,
+    operation: null,
+  });
 
-  componentDidMount() {
+  useEffect(() => {
     const btns = document.querySelectorAll('button');
     Array.from(btns).forEach((btn) => {
       btn.addEventListener('click', () => {
         const value = btn.getAttribute('data');
-        this.setState((currState) => calculate(currState, value));
+        callBack(calculate(state, value));
       });
     });
```

Comment:
- [ ] [OPTIONAL] Since you are working with `React` I think it would be better to  create an array to get each `button ` and loop through an array that may help you get all buttons and give the even click to one

Reply:
Thanks @hiromataba ,
I'll surely explore this suggestion and see how best I can implement it.